components/combobox/combobox.vue

```diff
@@ -317,6 +327,10 @@ export default {
 
       if (this.highlightIndex >= 0) {
         this.$emit('select', this.highlightIndex);
+
+        if (this.clickOnSelect) {
+          this.activeItemEl?.click();
```

Comment:
will this emit the `select` event two times? the second will be for this click

Reply:
My intention was that when using this option you wouldn't handle the selection from the combobox's `@select` event anymore, but just by attaching a `@click` handler on the list item. It still emits the event in case it's needed for some reason, but it's not needed anymore as long as the click handlers are there.

Reply:
If you think this is not clear enough and people might accidentally start using this option in conjunction with the `@select` event, which would admittedly result in double selections, we could consider changing this so that instead of a "select" event it emits a "clicked" or "item-clicked" event instead when this options is being used.

I think it's still important to emit something here even when using this option so that we can distinguish between keyboard and mouse selections in the parent component for amplitude tracking reasons for example.

Reply:
Ok, yeah I can't think in which case you'd use both, the `select` and `click` events.
I was confused at first and I thought it would emit the select twice but it's because we're using the same event handler for the click in the [SB's default story](https://github.com/dialpad/dialtone-vue/blob/staging/components/combobox/combobox_default.story.vue#L37). So updating this handler to log the `click` event in SB I think will remove the confusion.
Example:

```js
// combobox_default.story.vue
        <dt-list-item
          v-for="(item, i) in displayItems"
          :key="item.id"
          role="option"
          navigation-type="arrow-keys"
          @click="onItemClick(i)"
        >
```

```js
// combobox.stories.js
export const argsData = {
  ...
  onItemClick: action('click'),
};

export const argTypesData = {
  ...
  onItemClick: {
    table: {
      disable: true,
    },
  },

```

Reply:
Ah gotcha, yeah makes total sense. I'll add a different action name for the click like you suggested.